packages/user/src/utils.ts

```diff
@@ -0,0 +1,174 @@
+ * @param init Initial values for the request
+ * @returns The response body interpreted as JSON
+ */
+export async function requestAPI<T>(
+  endPoint = '',
+  init: RequestInit = {}
+): Promise<T> {
+  const settings = ServerConnection.makeSettings();
+  const requestUrl = URLExt.join(settings.baseUrl, endPoint);
+
+  let response: Response;
+  try {
+    response = await ServerConnection.makeRequest(requestUrl, init, settings);
+  } catch (error) {
+    throw new ServerConnection.NetworkError(error);
+  }
+
+  let data: any = await response.text();
+
+  if (data.length > 0) {
+    try {
+      data = JSON.parse(data);
+    } catch (error) {
+      console.log('Not a JSON response body.', response);
+    }
+  }
+
+  if (!response.ok) {
+    throw new ServerConnection.ResponseError(response, data.message || data);
+  }
+
+  return data;
+}
+
+// From https://en.wikipedia.org/wiki/Moons_of_Jupiter
+export const moonsOfJupyter = [
+  'Metis',
+  'Adrastea',
+  'Amalthea',
+  'Thebe',
+  'Io',
+  'Europa',
+  'Ganymede',
+  'Callisto',
+  'Themisto',
+  'Leda',
+  'Ersa',
+  'Pandia',
+  'Himalia',
+  'Lysithea',
+  'Elara',
+  'Dia',
+  'Carpo',
+  'Valetudo',
+  'Euporie',
+  'Eupheme',
+  // 'S/2003 J 18',
+  // 'S/2010 J 2',
+  'Helike',
+  // 'S/2003 J 16',
+  // 'S/2003 J 2',
+  'Euanthe',
+  // 'S/2017 J 7',
+  'Hermippe',
+  'Praxidike',
+  'Thyone',
+  'Thelxinoe',
+  // 'S/2017 J 3',
+  'Ananke',
+  'Mneme',
+  // 'S/2016 J 1',
+  'Orthosie',
+  'Harpalyke',
+  'Iocaste',
+  // 'S/2017 J 9',
+  // 'S/2003 J 12',
+  // 'S/2003 J 4',
+  'Erinome',
+  'Aitne',
+  'Herse',
+  'Taygete',
+  // 'S/2017 J 2',
+  // 'S/2017 J 6',
+  'Eukelade',
+  'Carme',
+  // 'S/2003 J 19',
+  'Isonoe',
+  // 'S/2003 J 10',
+  'Autonoe',
+  'Philophrosyne',
+  'Cyllene',
+  'Pasithee',
+  // 'S/2010 J 1',
+  'Pasiphae',
+  'Sponde',
+  // 'S/2017 J 8',
+  'Eurydome',
+  // 'S/2017 J 5',
+  'Kalyke',
+  'Hegemone',
+  'Kale',
+  'Kallichore',
+  // 'S/2011 J 1',
+  // 'S/2017 J 1',
+  'Chaldene',
+  'Arche',
+  'Eirene',
+  'Kore',
+  // 'S/2011 J 2',
+  // 'S/2003 J 9',
+  'Megaclite',
+  'Aoede',
+  // 'S/2003 J 23',
+  'Callirrhoe',
+  'Sinope'
+];
+
+/**
+ * Get a random user-name based on the moons of Jupyter.
+ * This function returns names like "Anonymous Io" or "Anonymous Metis".
+ */
+export const getAnonymousUserName = (): string =>
+  'Anonymous ' +
+  moonsOfJupyter[Math.floor(Math.random() * moonsOfJupyter.length)];
+
+/**
+ * Predefined colors for users
+ */
+export const userColors = [
+  '#12A0D3',
+  '#17AB30',
+  '#CC8500',
+  '#A79011',
+  '#ee6352',
+  '#609DA9',
+  '#4BA749',
+  '#00A1B3'
+];
+
+/**
+ * Get a random color from the list of colors.
+ */
+export const getRandomColor = (): string =>
+  userColors[Math.floor(Math.random() * userColors.length)];
+
+/**
+ * Extract the initials from the name of the user.
+ */
+export const getInitials = (name: string, familyName?: string): string => {
```

Comment:
Also, even if we have `givenName` and `familyName` in `ICurrentUser` it may be be clear how to build initials. Maybe it makes sense to have an `initials` field in `ICurrentUser` that can be set or which defaults to using the first initials of `givenName` and `familyName`.

Reply:
I added the `initials` field to the `ICurrentUser`. I'm keeping the function as convenient to extract the initials from `givenName` and `familyName` in multiple places for anonymous users without repeating the code. However, this is now done when fetching the user information so every organization can change the logic of creating the initials.

The logic of this function has into account composed names and names with two last names (which is the case of the Spanish culture). For example, "Jose Carlos" could be a `givenName` and "Gonzáles García" is the last name.
Having this in mind, this function will extract the following initials depending on the input:

- `Jose Carlos`, `Gonzáles García` -> `JG`
- `Jose Carlos`, `` -> `JC`
- `Jose`, `Gonzáles` -> `JG`
- `Jose`, `` -> `J`

Reply:
Does this handle non-latin names?

Reply:
It is meant to handle common names (name, last name) plus composed names.
I updated the table to clarify.
Do you have an example of the name you are worried about?

Reply:
I mean e.g. chinese, japanese, hebrew, arabic, etc names written in their native script. Common assumptions about latin names do not always translate to such scripts (e.g. picking and capitalizing the first character of some parts of the name, split by whitespace, might become completely intelligible, and maybe downright rude).

Reply:
A reasonable resource: https://www.w3.org/International/questions/qa-personal-names

Reply:
To be clear: It might be completely ok to leave this as it is right now. I was mainly trying to stimulate the questions "What do we do if the name is not in latin script? Will the code fail with an exception? Does the concept of initials make sense in those cases? What do we do if it does not?"

Reply:
Those are good questions. Thanks for raising them.
Unfortunately, I do not have the answer.
In a case not contemplated here, the code won't fail. It could be a case where this function returns initials that don't make sense in other languages.